### PR TITLE
chore: update `build.yml` to use `lean-action`

### PR DESCRIPTION
**Pull request by austinletson**
Replace the `install elan`,  `build batteries`, `test batteries`, and `lint batteries` steps of the Build job with `lean-action`.

### CODE REVIEW

**Comment by austinletson** [comment]
@semorrison what should we verify here before merging?

Based on the build output, `lean-action` seems to be doing the same steps as the step-by-step ci. I plan to take a closer look tomorrow morning with fresh eyes.

Additionally, I am happy to perform any `batteries` specific testing on this PR. What do you think?

**Comment by austinletson** [comment]
[Here](https://github.com/leanprover-community/batteries/actions/runs/9190636187/job/25275288278?pr=805) is an example of a build failure.

**Comment by austinletson** [comment]
[Here](https://github.com/leanprover-community/batteries/actions/runs/9190708090/job/25275529374?pr=805) is an example of a test failure.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -14,26 +14,13 @@ jobs:
...
-        run: lake build -Kwerror
-
-      - name: test batteries
-        if: steps.build.outcome == 'success'
-        run: lake test
-
-      - name: lint batteries
-        if: steps.build.outcome == 'success'
-        run: make lint
+      - id: lean-action
```
Comment (semorrison) [inline .github/workflows/build.yml:19]:
```suggestion
      - id: lean-action
      - name: build, test, and lint batteries
```

**Pull request merged**